BuildResidentialHPXML/resources/schedules.rb

```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 require 'csv'
-require 'yaml'
+require 'json'
```

Comment:
This change also shaves 0.2s.

Reply:
I really like having those comments alongside the values, so not sure if it is worth getting rid of yaml to save 0.2s.

Reply:
Yes, this is definitely the only controversial change. Indeed the lack of comments in json is unfortunate. But it felt like 0.2s is worth it, especially if A) we are running hundreds of thousands of simulations, or B) we ever want to use this for more use cases (like Home Energy Score) where we are doing everything we can to keep runtimes as fast as possible.

If we really need embedded comments, we could consider CSV or XML (both of which won't introduce overhead like YAML).

Reply:
Thanks! Looks like a case of we gotta lose something to gain something :)

Reply:
Also note that I did toy with using embedded comments in the JSON like so:
https://github.com/NREL/OpenStudio-HPXML/blob/d4e34809f996891ef64f52f0f23528c023bce24a/BuildResidentialHPXML/resources/schedules_config.json

Do you think that's better than the current implementation of using a separate file?

Reply:
I personally find the implementation of comments on JSON even more confusing than having a separate file.

How about adding a script to autogenerate the json from the yaml but keeping the yaml around?
The workflow would be:
1. Make modifications in the yaml.
2. Run the script to generate the JSON.
3. Commit both the json and yaml.

The risk is someone modifying the yaml but forgetting to run the script, thereby silently and mistakenly using the old config while believing they are using new one.

Reply:
Hmm. I can understand the motivation, but the programmer in me 😬s at the thought of redundant (and, as you point out, potentially out of sync) information.

Reply:
Hehe, I understand. It would have been nice if we could just convert it on the fly, but then that defeats the whole purpose and makes it even slower. I guess, it's a unique problem due to having to run this whole thing 1000s of times in succession. If there was a initial setup step in the workflow where we could do this just once before kicking off all the simulations, that would have been great.